README.md

```diff
@@ -1,18 +1,129 @@
+
+## Quickstart
+
+### The effect list
+
+The effect list `r` in the type `Eff r a` is a central concept in this library.
+It is a type-level list containing effect types. If `r` is ... then,
+
+If `r` is the empty list, then the computation `Eff r` (or `Eff '[]`) does not contain any effects to be handled and therefore is a pure computation. In this case, the result value can be retrieved by `run :: Eff '[] a -> a`
+
+For programming within the `Eff r` monad, it is never necessary to list all effects that can appear.
+It suffices to state what types of effects are at least required.
+This is done via the `Member t r` typeclass. It describes that the type `t` occurs inside the list `r`.
+
+Effect-handler functions, mostly called `run*`, `exec*` or `eval*`, typically have a signature like `Eff (x ': r) a -> Eff r a`. Where `x` is the type of effect to be handled, like `Writer w` or `State s`. The transformation from the longer list of effects `(x ': r)` to just `r` is a type-level indicator that the effect `x` has been handled. Depending on the effect, some additional input might be required or some output is produced.
+
+## Most common effects
+
+The most common effects used are `Writer`, `Reader`, `Exception` and `State`.
+
+Every effect has 2 important functions
+
+* how to generate the effect.
+* how the effect is done
+
+For the basic effects, the types of those functions clearly state the intent of the effect.
+
+In this section, the utility functions of the effects are not listed. Once you understand the basic usage of an effect, the type of any utility function typically explains what is does.
+
+Of course, the best learning progress is done by experimenting inside `ghci`.
+
+### The Exception Effect
+
+The exception effect adds the possibility to exit a computation preemptively with an exception. Note that the exceptions from this library are handled by the programmer and have nothing to do with exceptions thrown inside the Haskell run-time.
+
+```haskell
+throwError :: Member (Exc e) r => e -> Eff r a
+runError :: Eff (Exc e ': r) a -> Eff r (Either e a)
+```
+
+An exception can be thrown using the `throwError` function. Its return type is `Eff r a` with an arbitrary type `a`. When handling the effect, the result-type changes to `Either e a` instead of
+just `a`. This indicates how the effect is handled: The returned value is either the thrown exception or the value returned.
+
+### The State Effect
+
+The state effect provides readable and writable state during a computation.
+
+```haskell
+get :: Member (State s) r => Eff r s
+put :: Member (State s) r => s -> Eff r ()
+modify :: Member (State s) r => (s -> s) -> Eff r ()
```

Comment:
@power-fungus as per the pattern clarified above, `get`, `put` together form an algebra (encapsulated in the `State` GADT). each is a distinct kind of request we can make to the handler of the effect and the functions here embody each of the requests (each corresponding to a distinct case of the GADT).

`modify` is a combined operation and represents the combination of the `get` and `put` requests.

Reply:
I will mention modify as a combination of get and put.

I get that the emphasis on the constructed effect-algebra is an important part of the library concept. However, it can be used without its knowledge by just treating them as normal effectful, lazy functions. For that reason, I do not want it to be introduced in the Quickstart.

Maybe it is a good fit when describing how to implement custom effects. Let's see.

Reply:
@power-fungus agreed - this would be relevant for custom effects, not for using existing ones